src/precice/impl/SolverInterfaceImpl.cpp

```diff
@@ -1238,54 +1212,15 @@ void SolverInterfaceImpl::readBlockScalarData(
     const int *valueIndices,
     double *   values) const
 {
-  PRECICE_TRACE(dataID, size);
-  PRECICE_CHECK(_state != State::Finalized, "readBlockScalarData(...) cannot be called after finalize().");
-  PRECICE_VALIDATE_DATA_ID(dataID);
-  if (size == 0)
-    return;
-  PRECICE_DEBUG("size = " << size);
-  PRECICE_ASSERT(valueIndices != nullptr);
-  PRECICE_ASSERT(values != nullptr);
-  PRECICE_REQUIRE_DATA_READ(dataID);
-  DataContext &context = _accessor->dataContext(dataID);
-  PRECICE_ASSERT(context.toData != nullptr);
-  mesh::Data &data = *context.toData;
-  PRECICE_CHECK(data.getDimensions() == 1,
-                "You cannot call readBlockScalarData on the vector data type \"" << data.getName()
-                                                                                 << "\". Use readBlockVectorData or change the data type for \"" << data.getName() << "\" to scalar.");
-  auto &     valuesInternal = data.values();
-  const auto vertexCount    = valuesInternal.size();
-  for (int i = 0; i < size; i++) {
-    const auto valueIndex = valueIndices[i];
-    PRECICE_CHECK(0 <= valueIndex && valueIndex < vertexCount, "Cannot read data \"" << data.getName() << "\" to invalid Vertex ID (" << valueIndex << "). Please make sure you only use the results from calls to setMeshVertex/Vertices().");
-    values[i] = valuesInternal[valueIndex];
-  }
+  readBlockVectorDataImpl<1>(dataID, size, valueIndices, values);
```

Comment:
Why not simplifying the name to `readBlockDataImpl`?

Reply:
We don't really need dimensions here. What's important is weather the user called the scalar or vector variant.

Reply:
I fixed it